packages/react-magma-dom/src/components/StyledButton/styles.ts

```diff
@@ -297,28 +367,45 @@ export function buildActiveColor(props) {
   props = buildPropsWithDefaultButtonStyles(props);
 
   if (props.color === 'marketing') {
+    if (props.variant !== 'solid' && props.isInverse) {
+      return props.theme.colors.tertiary500;
+    }
     return props.theme.colors.foundation02;
   }
+  if (props.color === 'danger') {
+    if (props.variant !== 'solid' && !props.isInverse) {
+      return darken(0.2, props.theme.colors.danger);
+    }
+  }
   if (
     (props.variant !== 'solid' && !props.isInverse) ||
     (props.variant === 'solid' && props.isInverse)
   ) {
     switch (props.color) {
       case 'secondary':
-        return props.theme.colors.neutral;
-      case 'success':
-        return darken(0.2, props.theme.colors.success);
+        return darken(0.2, props.theme.colors.primary);
       case 'danger':
-        return darken(0.2, props.theme.colors.danger);
+        return props.theme.colors.neutral100;
       default:
         return darken(0.2, props.theme.colors.primary);
     }
   }
   if (props.variant !== 'solid' && props.isInverse) {
-    return props.theme.colors.neutral08;
+    switch (props.color) {
+      case 'primary':
+        return props.theme.colors.tertiary500;
+      case 'secondary':
+        return props.theme.colors.tertiary500;
+      case 'danger':
+        return props.theme.colors.danger200;
+    }
+  }
+
+  if (props.variant !== 'solid' && props.isInverse) {
+    return props.theme.colors.neutral100;
   }
```

Comment:
This should be condensed as is a duplicate of the if statement on line 393

Reply:
Done